File: 0633/solution.java
class Solution {
    public boolean judgeSquareSum(int c) {
        int root = (int) Math.sqrt(c);
        for (int a = 0; a <= root; a++) {
            double b = Math.sqrt(c - (a * a));
            if (b == (int) b) {
                return true;
            }
        }
        return false;
    }
}

// time O(sqrt(c)log(c))
// space O(1)

File: 0633/solution.py
from math import sqrt


class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        for i in range(int(sqrt(c)) + 1):
            b = c - (i**2)
            root = int(sqrt(b))
            if root * root == b:
                return True
        return False
